acpreprocessing/stitching_modules/convert_to_n5/tiff_to_ngff.py

```diff
@@ -598,16 +632,243 @@ def write_mimgfns_to_n5(
+        chunk size for n5 datasets
+    concurrency : int
+        total concurrency used for writing arrays to n5
+        (python threads = concurrency // slice_concurrency)
+    slice_concurrency : int
+        threads used by z5py
+    compression : str, optional
+        compression for n5 (default: raw)
+    dtype : str, optional
+        dtype for n5 (default: uint16)
+    lvl_to_mip_kwargs :  dict, optional
+        mapping of MIP level to kwargs used in MIPmap generation
+    interleaved_channels : int
+        number of channels interleaved in the tiff files (default 1)
+    channel : int, optional
+        channel from which interleaved data should be read (default 0)
+    deskew : str, optional
+        str id for parameters to run pixel shifting deskew (default '')
+    """
+    group_attributes = ([] if group_attributes is None else group_attributes)
+    
+    joined_shapes = joined_mimg_shape_from_fns(
+        mimgfns, concurrency=concurrency,
+        interleaved_channels=interleaved_channels, channel=channel)
+    # TODO also get dtype from mimg
+    # TODO KT DESKEW: reshape joined_shapes to deskewed dimensions
+    if deskew:
+        deskew_options = psd.options_from_str(deskew)
+        block_size = chunk_size[2]
+        slice_length = int(chunk_size[2]/deskew_options['stride'])
+        deskew_kwargs = psd.psdeskew_kwargs(skew_dims_zyx=(slice_length,joined_shapes[1],joined_shapes[2]),
+                                            **deskew_options
+                                            )
+        joined_shapes = psd.reshape_joined_shapes(joined_shapes,**deskew_kwargs)
+        #print('deskewed joined shapes is ' + str(joined_shapes))
+    else:
+        block_size = chunk_size[2]
+        slice_length = block_size
+        deskew_kwargs = {}
+
+    # TODO DESKEW: does this generally work regardless of skew?
+
+    workers = concurrency // slice_concurrency
+    
+    zstore = zarr.DirectoryStore(output_n5,dimension_separator='/')
+    with zarr.open(zstore,mode='a') as f:
+        mip_ds = {}
+        # create groups with attributes according to omezarr spec
+        if len(group_names) == 1:
+            group_name = group_names[0]
+            try:
+                g = f.create_group(f"{group_name}")
+            except KeyError:
+                g = f[f"{group_name}"]
+            try:
+                attributes = group_attributes[0]
+            except IndexError:
+                print('attributes error')
+                
+            if "pixelResolution" in attributes:
+                if deskew:
+                    attributes["pixelResolution"]["dimensions"][2]/=deskew_options["stride"]
+                attributes = omezarr_attrs(group_name,attributes["position"],attributes["pixelResolution"]["dimensions"],max_mip)
+            if attributes:
+                for k, v in attributes.items():
+                    g.attrs[k] = v
+        else:
+            print('omezarr should only have one group')
```

Comment:
Do you want to elevate this to an exception?  If not, we should figure out what log level this should be once #46 is resolved

Reply:
Elevating sounds right.